### ISSUE TITLE
Unable to read tests class using the Revit version 2024

### ISSUE THREAD

**Issue by FabianOswald-WolfSystem** [ricaun-io/RevitTest#17]
I get the following error as soon as I have a project reference to another non-test project in my test project.

OutputConsole: -------------------------------
OutputConsole: ricaun.RevitTest.Command 1.5.0 [Net Framework] [ricaun.NUnit 1.4.0 3.13.3]
OutputConsole: -------------------------------
OutputConsole: RevitTestUtils: 2024
OutputConsole: RevitTestUtils: C:\Program Files\Autodesk\Revit 2024\
OutputConsole: System.Exception: Unable to read tests class using the Revit version 2024.
OutputConsole: bei ricaun.RevitTest.Console.Revit.Utils.RevitTestUtils.GetTestFullNames(String filePath)
OutputConsole: bei ricaun.RevitTest.Command.RunCommand`1.RunOrReadTests()
OutputConsole: bei ricaun.RevitTest.Command.RunCommand`1.Run()
OutputConsole: ERROR: Unbehandelte Ausnahme: System.Exception: Unable to read tests class using the Revit version 2024.
OutputConsole: ERROR:    bei ricaun.RevitTest.Console.Revit.Utils.RevitTestUtils.GetTestFullNames(String filePath)
OutputConsole: ERROR:    bei ricaun.RevitTest.Command.RunCommand`1.RunOrReadTests()
OutputConsole: ERROR:    bei ricaun.RevitTest.Command.RunCommand`1.Run()
OutputConsole: ERROR:    bei CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
OutputConsole: ERROR:    bei ricaun.RevitTest.Command.RunTest.ParseArguments[T](String[] args)
OutputConsole: ERROR:    bei ricaun.RevitTest.Console.Program.Main(String[] args)
DiscoverTests: -------------------------------
DiscoverTests: []
DiscoverTests: -------------------------------
DiscoverTests: -------------------------------

Strangely enough, the error does not occur on one computer and the tests are all run. This error occurs on two other computers. Unfortunately, I cannot provide you with a project for reproduction.

**Comment by FabianOswald-WolfSystem** [comment]
I have looked into the problem again. As soon as the DLL file msvcr100.dll is included or is only located in the output directory, the error message in question appears. The msvcr100.dll DLL file is part of the Microsoft Visual C++ 2010 Redistributable and contains runtime functions that are used by applications developed in Microsoft Visual C++ 2010. We access C++ functions to communicate with our ERP system. I have attached the dll.
[msvcr100.zip](https://github.com/user-attachments/files/17160646/msvcr100.zip)

**Comment by ricaun** [comment]
What version do you have installed in the machine? 

Because the discovery tries to load the original Revit dll files found the folder, usually tries to use version 2021, 2022 or 2023 reference to fix some popup problem. 

This would explain why in one machine works and another don't. If you only have version 2024 installed.

**Comment by ricaun** [comment]
> I have looked into the problem again. As soon as the DLL file msvcr100.dll is included or is only located in the output directory, the error message in question appears. The msvcr100.dll DLL file is part of the Microsoft Visual C++ 2010 Redistributable and contains runtime functions that are used by applications developed in Microsoft Visual C++ 2010. We access C++ functions to communicate with our ERP system. I have attached the dll. [msvcr100.zip](https://github.com/user-attachments/files/17160646/msvcr100.zip)

Don't know what is this file but I don't think that is related with the issue.

One thing you could do is make all class that is not for test private. 

If you run `dotnet test` the tests run? This would confirm the problem is on the discovery tests.

**Comment by ricaun** [comment]
Here is a way to understand and test if your lowest version of Revit is 2024.

If you add this class in a test file that is working fine.
```
public class Command : IExternalCommand
{
    public Result Execute(ExternalCommandData commandData, ref string message, ElementSet elementSet)
    {
        return Result.Succeeded;
    }
}
```
When the discovery of test execute each class gonna be loaded and in this case the RevitAPIUI gonna be loaded because of the `IExternalCommand` interface, and that is not allow in some versions of Revit like version 2024.

If you change the class to internal, and because you are using version 1.5.* that gonna retry to find all the class that is only public, ignoring the `Command` class and `RevitAPIUI` is not gonna be loaded in the discovery.

I forgot to add the exception related in the log related to the `GetTypes` failing.

**Comment by ricaun** [comment]
I'm doing some experiments if you use the last prerelease version `1.5.1-beta.3` I add a fake RevitAPIUI and this would fix the issue. If you could test I appreciated.

**Comment by FabianOswald-WolfSystem** [comment]
Thank you very much for the extensive feedback. I have tested the prerelease version intensively and can confirm that it works. The problem that Revit 2023 must be installed is therefore no longer an issue. I was able to run the test cases on several test machines.

**Comment by ricaun** [comment]
> Thank you very much for the extensive feedback. I have tested the prerelease version intensively and can confirm that it works. The problem that Revit 2023 must be installed is therefore no longer an issue. I was able to run the test cases on several test machines.

Great, thanks for testing!

Eventually I gonna update to version 1.6 with this change 😀

**Comment by ricaun** [comment]
The version 1.6.0 is released! 
Now is not necessary to have some random Revit version installed to discovery the tests. The discovery gonna use a mock/fake version of the RevitAPIUI to be used as reference just to find the tests names.

This update probably fix this issue: #9

**Comment by ricaun** [comment]
This is a sample code just to show that the RevitTest version 1.6.0 is using a mock `RevitAPIUI`
```
public class Tests_Discovery
{
    public static IEnumerable<string> GetAssemblies()
    {
        var types = new[] { 
            typeof(UIFramework.RevitRibbonControl),
            typeof(Autodesk.Windows.RibbonControl),
            typeof(Autodesk.Revit.UI.UIApplication),
            typeof(Autodesk.Revit.DB.Document)
        };
        return types.Select(t => t.Assembly.ToString());
    }

    [TestCaseSource(nameof(GetAssemblies))]
    public void TestAssemblyName(string assembly)
    {

    }
}
```
